pkg/core/blockchain.go

```diff
@@ -925,6 +936,23 @@ func (bc *Blockchain) LastBatch() *storage.MemBatch {
 	return bc.lastBatch
 }
 
+// removeOld removes all blocks and transactions from start to stop height inclusively.
+func (bc *Blockchain) removeOld(d dao.DAO, h util.Uint256) error {
+	b, err := bc.GetBlock(h)
+	if err != nil {
+		return err
+	}
+	for _, tx := range b.Transactions {
+		if err := d.DeleteTransaction(tx.Hash()); err != nil {
+			return err
+		}
+	}
```

Comment:
You're not accounting for execution results BTW.

Reply:
Fixed.